### ISSUE TITLE
cockpit ignores ListenAddress setting

### ISSUE THREAD

**Issue by hashworks** [cockpit-project/cockpit#2052]
I've followed the [guide](http://files.cockpit-project.org/guide/listen.html) to change the ListenStream and ListenAddress of cockpit:
```
[Socket]
ListenStream=5443
ListenAddress=127.0.0.1
```
The ListenStream setting works fine (altrough I had to remove it from `/lib/systemd/system/cockpit.socket`, otherwise it would listen on `9090` and `5443`), however cockpit keeps ignoring the ListenAddress setting, it is still accessible on `0.0.0.0`.

**Comment by stefwalter** [comment]
ListenAddress is not a valid systemd setting. I've prepared a patch that removes it from our documentation.

**Comment by stefwalter** [comment]
I think you can use:

```
ListenStream 127.0.0.1:5443
```

But you should see systemd documentation for more information.

**Comment by hashworks** [comment]
Don't forget the `=`:
`[/etc/systemd/system/cockpit.socket.d/listen.conf:2] Missing '='.`

Using `ListenStream=127.0.0.1:5443` works fine, thanks! Listening on localhost and 5443 only.

However cockpit-ws still tells me `HTTP Server listening on port 9090`, I'll open another issue for that one.

**Comment by mvollmer** [comment]
Fixed with e97eec6c487636e0

**Comment by jeffli678** [comment]
@hashworks As a minor suggestion, You should NOT have deleted the /lib/systemd/system/cockpit.socket. To stop cockpit from listening on both ports, keep the first empty line ListenStream= exactly as in the guide. 

The config file in /lib/systemd/system says the cockpit should listen on port 9090. In your own config file in /etc/systemd/system, you use the first empty line to override the port 9090 in /lib/..., and set a new port  (in the second line) for which the cockpit listens on. I know it is a little bit confusing, but it is the supposed way. 

The /lib/systemd/system config may be unexpectedly restored upon upgrade. So you could end up exposing your cockpit on port 9090 silently.
